### PR TITLE
Remove custom robots.txt in favor of RTD default

### PR DESCRIPTION
**Pull request by DimedS**
## Description
This PR removes the custom `robots.txt`. The file automatically generated by Read the Docs (RTD) will be used during the build.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -214,7 +214,8 @@
 
 # html_extra_path used to define a path to robots.txt which is used by webcrawlers
 # to ignore or allow certain links.
-html_extra_path = [str(here / "robots.txt")]
+# We've decided to rely on RTD's default robots.txt, so we no longer need to specify a custom one
+# html_extra_path = [str(here / "robots.txt")]
```
Comment (astrojuanlu) [inline docs/source/conf.py:218]:
```suggestion
# We've decided to rely on RTD's default robots.txt, so we no longer need to specify a custom one,
# see discussion at https://github.com/kedro-org/kedro/issues/3741
# html_extra_path = [str(here / "robots.txt")]
```

**Review by astrojuanlu — APPROVED** [review]
It will be difficult to test this, but I'd just say let's go for it.

**Comment by noklam** [comment]
Is this ready?

**Comment by DimedS** [comment]
> Is this ready?

This PR is ready, but the entire SEO optimisation isn't complete yet. If you're asking whether we should merge it before the release, I believe it's better to finish all the optimisations together. What do you think, @astrojuanlu?

**Comment by astrojuanlu** [comment]
If I understand correctly, removing `robots.txt` will only increase the amount of URLs we're indexing, right? So maybe we can go ahead and try the sitemap separately? https://github.com/kedro-org/kedro/issues/3741#issuecomment-2281066853

**Review by noklam — APPROVED** [review]

**Pull request merged**